fix: remove `/suggestions/api/4_1/rs/suggest/` path for the customEnd…

nameless19922 opened the pull request:
It's necessary to pass `customEndpoint` without `/suggestions/api/4_1/rs/suggest/` part.
For example, I use proxy api url `https://example.com/api/suggestion/fio`

Commit:
fix: remove `/suggestions/api/4_1/rs/suggest/` path for the customEndpoint

RUVATA commented on dist/index.js:
```diff
@@ -371,7 +371,7 @@ var _initialiseProps = function _initialiseProps() {
     }
 
-    _this3.xhr.open('POST', (customEndpoint && customEndpoint.slice(-1) === '/' ? customEndpoint.slice(0, -1) : customEndpoint || 'https://suggestions.dadata.ru') + '/suggestions/api/4_1/rs/suggest/' + type);
+    _this3.xhr.open('POST', (customEndpoint && customEndpoint.slice(-1) === '/' ? customEndpoint.slice(0, -1) : customEndpoint || 'https://suggestions.dadata.ru/suggestions/api/4_1/rs/suggest') + '/' + type);
```
I think that i understood the problem, but this - changed nothing, logic of string concatination are equal, at before to after your change's

RUVATA reviewed (commented):
@nameless19922
> It's necessary to pass `customEndpoint` without `/suggestions/api/4_1/rs/suggest/` part.
> For example, I use proxy api url `https://example.com/api/suggestion/fio`

I understood the problem, but I propose to solve it in a different way.
We can make to customEndpoint prop - custom definition as object, thats defaults
`
{
 host: 'https://suggestions.dadata.ru/'
 api: 'suggestions/api/4_1/rs/suggest'
}
`
When it needed  - user can setup it manually - to set own custom host and own custom api call.
What are you think about that ?

nameless19922 commented:
@RUVATA 
It will be just right. Thx.

Commit:
closes #25 issue
Possibility to setup custom API part of customEndpoint usage

closes #25 issue (refactoring)
Refactoring to resolve breaking changes of first implementation

closes #25 issue (build)
Refactoring to resolve breaking changes of first implementation

closes #25 issue (refactor)
Fix Braking changes - relative path support

closes #25 issue (fix)
Fix Braking changes - relative path support

RUVATA commented:
@nameless19922 
> It will be just right. Thx.

i'am opened issue #25 based on your improve idea
and pushed commit [049d7f1](https://github.com/orbita-center/react-dadata-box/pull/24/commits/049d7f1828f0e50bad3c9f2b8b578044c917493e) with implementation of my proposal to your fork branch.
Please see then, are that closes that what are you need ?

If it's all right - i'am test there and merge to v1.2.9 and publish it in nearest release to NPM

RUVATA commented:
@nameless19922
[73f7198](https://github.com/orbita-center/react-dadata-box/pull/24/commits/73f7198f242664e61386f8904bb98c3611436994) is refactoring of first implementation, that are make breaking changes to use **customEndpoint** property
thats may be used already, based on logic in that places only host but api-path (suggestions/api/4_1/rs/suggest) will added automatically

PS: after that we find finally decision - we will squash commits in candidate branch

RUVATA commented:
[closes orbita-center#25 issue...](https://github.com/nameless19922/react-dadata-box/commit/6dcb5fa739843690a29da14344d4d368282c7778) - squash commits

Commit:
closes #25 issue (README)

nameless19922 reviewed (commented):
@RUVATA 

Ok. It works as it should.

RUVATA commented:
@nameless19922 
own work assimilated to #26 and wait approve of other contributors.

RUVATA commented:
@nameless19922 
At now - v1.2.9 with a requested functionality is published on NPM
Please update your react-dadata-box package.
Thank you for your usage, contribution and attention to our solution.

RUVATA commented:
@nameless19922
after v1.2.9, i'am understood that i forgot update JSDoc & Types Definition, and later when we are refactored code in own project for use new definition variant of customEndpoint property - i'am found trouble with generate target URI when property 'api' in object is skipped.
Now we released 1.3.1 (#29) version - where resolved this trouble and fixed JSDoc and TypeScript typing - welcome to update.
